### ISSUE TITLE
All my notes are missing

### ISSUE THREAD

**Issue by jjaimon** [HappenApps/Quiver#686]
I've been using Quiver for the last 10 months.  I'm on v3.0.3(88) currently. I haven't opened Quiver ever since I started upgrading to macOS Sierra PBs, ie. since July.  I just realized that Quiver lost all my notes.  The `Quiver.qvlibrary` file in the default location (`$HOME/Library/Containers/com.happenapps.Quiver/Data/Library/Application Support/Quiver/`) seems to be a new file with just 84 bytes.  From the timiestamp and **Timemachine** backups, I figured that this file was created in April.  I checked TImemachine backups all the way to December and I can't see any qvlibrary file in this location.  I also inspected `settings.json` in the same location to make sure that I haven't relocated the library to elsewhere.  

Unfortunately I didn't maintain a backup for this.  Is there way to figure out how I lost all the files and retrieve from Timemachine?  Were there any alternate location in the previous versions?  

Any help is very much appreciated as I really want to retrieve my notes.

**Comment by jjaimon** [comment]
I changed the library location in Preferences->Sync.  I don't see that change reflecting in `settings.json`. So, certainly there is another location where some settings are stored.

**Comment by ylian** [comment]
From your description, it seems that you moved your library elsewhere. Somehow that library is gone, so when you open Quiver again, it creates a new library at the default location, and at the same time updated `settings.json`.

I would suggest you looking for your library in some common places: Dropbox, other cloud folders, etc. If you still can't find it, maybe do a global search in the terminal:

```
find ~ -name 'Quiver.qvlibrary' -print
```

**Comment by jjaimon** [comment]
I don't remember moving the library anywhere.  Where is this information (location of library) stored?  I expected this to be in `settings.json` and tried to validate my assumption by moving the current library elsewhere.  I don't see the new location updated in `settings.json`.  This information will help me restore from Timemachine backup.

**Comment by jjaimon** [comment]
I searched the entire home directory.  I don't see this file anywhere. :-(

**Comment by ylian** [comment]
The library location is stored as "bookmark data" in `library.plist`, which is in the same directory as `settings.json`. But I'm afraid this file is already overwritten (to point to the default location) when you opened Quiver. The old location is not saved there any more.

Maybe try use Time Machine to go back a few months, and search in Finder for "Quiver.qvlibrary".

**Comment by jjaimon** [comment]
Thank you very much. This helped.  I had indeed moved the library location.  The reason asking for the settings file was to look up the Timemachine for the old file and retrieve the library location.  I tried to locate the settings by changing the library location and searching through out the Container to see where the new location is mentioned.  I didn't realize that the data itself is base64 encoded.  Long story short, I could locate the qvlibrary location and retrieve old file from my Timemachine backup.  Thanks again.
